src/sources/splunk_hec.rs

```diff
@@ -416,9 +426,9 @@ impl<R: Read> EventIterator<R> {
             log.insert(CHANNEL, guid.clone());
         }
 
-        // Process remote_addr field from X-Forwarded-For header
+        // Add host field from X-Forwarded-For header, if present.
         if let Some(splunk_remote_addr) = self.remote_addr.as_ref() {
-            log.insert(REMOTE_ADDR, splunk_remote_addr.clone());
+            log.insert(log_schema().host_key(), splunk_remote_addr.clone());
         }
```

Comment:
This won't quite work, since `remote: Option<SocketAddr>,` can overwrite it. Instead it can be added up above in `EventIterator` `fn new` `remote.map(|addr| addr.to_string()).map(Value::from)`. If `host` isn't present in payload it will use this value, so something like this will work 
```
remote_addr.or(remote.map(|addr| addr.to_string())).map(Value::from)
```

Reply:
Makes sense I think! That way it will set `remote_addr` to the xff's address first, then if not present assign to `remote: Option<SocketAddr>`.

Looking at this again, I _am_ a little confused as to the priority that the `EventIterator.extractors` would be applied here:
https://github.com/timberio/vector/blob/master/src/sources/splunk_hec.rs#L458-L461

Mostly just because my first read through `EventIterator::build_event` led me to think that the extractors would assign they're corresponding fields(`host` being one of them) after the x-forwarded-for logic I'm adding in here.

Is that read correct or am I missing what the `extractors` section of the code is supposed to do?

Sorry for all the questions!

Reply:
No problem. 

> the extractors would assign they're corresponding fields(host being one of them) after the x-forwarded-for logic

Yes, and if the event contains `host` it will overwrite before added xff, but if the event doesn't contain `host` it will set it as `remote: Option<SocketAddr>` which will overwrite xff if both are present. 

The problematic scenario would look like:
1. xff is present and is seted as `host`.
2. Events don't contain `host` so it isn't written.
3. Remote is present and 2. is true so it will overwrite `host`.

To solve this we can either check for this case as I mentioned in first comment, or we can change how extractors work which is a more complex change.

Reply:
A, I see, that was the misunderstanding, I meant this

```
         EventIterator {
            data,
            events: 0,
            channel: channel.map(Value::from),
            time: Time::Now(Utc::now()),
            extractors: [
                DefaultExtractor::new_with(
                    "host",
                    log_schema().host_key(),
                    remote_addr.or(remote.map(|addr| addr.to_string())).map(Value::from),
                ),
                DefaultExtractor::new("index", &INDEX),
                DefaultExtractor::new("source", &SOURCE),
                DefaultExtractor::new("sourcetype", &SOURCETYPE),
            ],
        }
```

So `DefaultExtractor` for `host` will set `remote_addr` for us if Events don't have it.

Reply:
Ahhhhhh ok, that makes a lot more sense -- ok; thank you for clarifying! I'll go ahead and fix that